src/MKL.jl

```diff
@@ -37,7 +72,7 @@ function set_interface_layer(interface::Interface = INTERFACE_LP64)
 end
 
 function __init__()
-    if MKL_jll.is_available()
+    if !(mkl_path == "mkl_jll" && !MKL_jll.is_available())
```

Comment:
Perhaps it would be less confusing to globally set a `mkl_found` that is set via `const mkl_found = MKL_jll.is_available()` when it's a JLL, and set to `libmklrt != ""` in the `system` case.